### PR TITLE
Switch to the Vanniktech base plugin for publishing

### PR DESCRIPTION
**Pull request by swankjesse**
Move configuration out of build.gradle files and into build.gradle.kts files.
Sign published builds.
Support publishing release builds from GitHub actions.

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -26,10 +26,13 @@ include(":moshi")
 include(":moshi:japicmp")
 include(":moshi:records-tests")
 include(":adapters")
+project(":adapters").name = "moshi-adapters"
```
Comment (swankjesse) [inline settings.gradle.kts:29]: In a follow-up I’d like to move the files around so we don’t need this renaming hack.

File: build.gradle.kts [hunk 2]
```diff
@@ -106,30 +112,54 @@ subprojects {
...
+      publishToMavenCentral(SonatypeHost.DEFAULT)
+      signAllPublications()
+      pom {
+        description.set("A modern JSON API for Android and Java")
+        name.set(project.name)
+        url.set("https://github.com/square/moshi/")
+        licenses {
+          license {
+            name.set("The Apache Software License, Version 2.0")
+            url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
```
Comment (ZacSweers) [inline build.gradle.kts:150]:
```suggestion
            url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
```

File: build.gradle.kts [hunk 3]
```diff
@@ -106,30 +112,54 @@ subprojects {
...
-        outputDirectory.set(rootDir.resolve("docs/1.x"))
-        dokkaSourceSets.configureEach {
-          skipDeprecated.set(true)
-          externalDocumentationLink {
-            url.set(URL("https://square.github.io/okio/2.x/okio/"))
+  plugins.withId("com.vanniktech.maven.publish.base") {
+    configure<MavenPublishBaseExtension> {
+      publishToMavenCentral(SonatypeHost.DEFAULT)
+      signAllPublications()
+      pom {
```
Comment (ZacSweers) [inline build.gradle.kts:143]: FYI there's a helper function for loading these from the same properties as before if ya want

**Review by ZacSweers — APPROVED** [review]

File: build.gradle.kts (hunk 3 above)
Comment (swankjesse) [reply build.gradle.kts:143]: Things I hate from most to hate least:

 * Gradle Groovy builds. Many ways to do each thing. Mistakes fail silently. Not discoverable.
 * Gradle properties files. One way to do each thing. Fails silently. Not discoverable.
 * Gradle Kotlin builds. IntelliJ tells me when I make a mistake. Discoverable.

So configuring stuff without the safety net of a type checker? Not unless I have no other choice.

**Pull request merged**